packages/expo/src/experimental/LocalCredentials.tsx

```diff
@@ -0,0 +1,150 @@
+import { useClerk, useSignIn } from '@clerk/clerk-react';
+import type { SignInResource } from '@clerk/types';
+import { AuthenticationType, isEnrolledAsync, supportedAuthenticationTypesAsync } from 'expo-local-authentication';
+import * as SecureStore from 'expo-secure-store';
+import type { PropsWithChildren } from 'react';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+type LocalCredentials = {
+  identifier: string;
+  password: string;
+};
+
+type BiometricType = 'fingerprint' | 'face-recognition';
+
+export const LocalAuthContext = createContext<{
+  setCredentials: (creds: LocalCredentials) => Promise<void>;
+  hasCredentials: boolean;
+  clearCredentials: () => void;
+  authenticate: () => Promise<SignInResource>;
+  biometryType: BiometricType | null;
+}>({
+  // @ts-expect-error Initial value cannot return what the type expects
+  setCredentials: () => {},
+  hasCredentials: false,
+  clearCredentials: () => {},
+  // @ts-expect-error Initial value cannot return what the type expects
+  authenticate: () => {},
+  biometryType: null,
+});
+
+export const useLocalCredentials = () => {
+  return useContext(LocalAuthContext);
+};
+
+const useEnrolledBiometric = () => {
+  const [isEnrolled, setIsEnrolled] = useState(false);
+
+  useEffect(() => {
+    let ignore = false;
```

Comment:
❓Is the purpose of `ignore` to not re-fetch `isEnrolledAsync`?

Reply:
So the intention is to avoid race conditions in a situation where the component gets unmounted but the promise resolves after the component is already unmounted.